File: Advanced Problems/Question 12/perfect_square.js
function perfect_square(num){
    for(let i = 1; i< num ; i++){
      if((num % i === 0) && (num / i === i)){
        return true;
      }
    }
    return false;
  }
  
  console.log(perfect_square(35));
  
  

File: Advanced Problems/Question 12/perfect_square.rb
def perfect_square?(num)
    (1..num).each do |factor|
      if factor * factor == num
        return true
      end
    end
  
    return false
  end
  
  puts perfect_square?(5)   #=> false
  puts perfect_square?(12)  #=> false
  puts perfect_square?(30)  #=> false
  puts perfect_square?(9)   #=> true
  puts perfect_square?(25)  #=> true